src/transformer/create-appendix.ts

```diff
@@ -0,0 +1,77 @@
+import url from 'node:url';
+
+import MagicString from 'magic-string';
+
+import type { AddonASTNodes, StoriesFileMeta } from 'src/parser/types.js';
+
+const parserModulePath = url
+  .fileURLToPath(new URL('../../dist/parser/collect-stories.js', import.meta.url))
+  .replace(/\\/g, '\\\\'); // For Windows paths
+
+export function createAppendix({
+  componentName,
+  code,
+  storiesFileMeta,
+  nodes,
+}: {
+  componentName: string;
+  code: MagicString;
+  storiesFileMeta: StoriesFileMeta;
+  nodes: AddonASTNodes;
+}) {
+  const { stories } = storiesFileMeta;
+  const { defineMetaVar } = nodes;
+  const parsedStoriesVariable = '__parsed';
+  const exportsOrderVariable = '__namedExportsOrder';
+
+  // NOTE:
+  // We need to remove the default export from the code,
+  // because Storybook internally expects export default `meta`
+  code.replace(/export default/, '');
+
+  const exportsOrder = Object.entries(stories).map(([id, _]) => id);
+  // biome-ignore format: Stop
+  // prettier-ignore
+  const storiesExports = Object.entries(stories)
+		.map(([id]) =>
+      `export const ${sanitizeStoryId(id)} = ${parsedStoriesVariable}.stories[${JSON.stringify(id)}];`
+		)
+		.join("\n");
+
+  // biome-ignore format: Stop
+  // prettier-ignore
+  const appendix = [
+    "", // NOTE: Adds a new line at the end of the code
+    `import parser from '${parserModulePath}';`,
+    `const ${parsedStoriesVariable} = parser(${componentName}, ${JSON.stringify(storiesFileMeta)}, ${findMetaPropertyName(defineMetaVar) ?? "meta"});`,
+    `export default ${parsedStoriesVariable}.meta;`,
+    `export const ${exportsOrderVariable} = ${JSON.stringify(exportsOrder)};`,
+    storiesExports,
+  ].join("\n");
+
+  code.append(appendix);
+
+  return code;
+}
+
+// FIXME: There's probably some interal function inside the Storybook to handle it
```

Comment:
I need help there. This is a temporary solution. I suspect there's some internal function in `storybook` to handle it?

This is to prevent story `id` to produce invalid syntax for variable name, it cannot have spaces nor special characters.